packages/x-charts/src/hooks/useInteractionItemProps.ts

```diff
@@ -18,18 +19,40 @@ export const useInteractionItemProps = (skip?: boolean) => {
       }
     };
     const onPointerEnter = () => {
-      dispatchInteraction({
-        type: 'enterItem',
-        data,
-      });
+      store.update((prev) => ({
+        ...prev,
+        interaction: {
+          ...prev.interaction,
+          item: data,
+        },
+      }));
       setHighlighted({
         seriesId: data.seriesId,
         dataIndex: data.dataIndex,
       });
     };
     const onPointerLeave = (event: React.PointerEvent) => {
       event.currentTarget.releasePointerCapture(event.pointerId);
-      dispatchInteraction({ type: 'leaveItem', data });
+
+      store.update((prev) => {
+        const prevItem = prev.interaction.item;
+        if (
+          prevItem === null ||
+          Object.keys(data).some(
+            (key) => data[key as keyof typeof data] !== prevItem[key as keyof typeof prevItem],
+          )
```

Comment:
I hate this about some native types, we should be able to override the output of the functions with something like: `Object.keys<keyof SeriesItemIdentifier>(data)` when the types don't work 🙃